test/parallel/test-debug-brk.js

```diff
@@ -1,9 +1,37 @@
 'use strict';
 
 const common = require('../common');
-const assert = require('assert');
-const spawnSync = require('child_process').spawnSync;
+const spawn = require('child_process').spawn;
 
-const args = [`--debug-brk=${common.PORT}`, '-e', '0'];
-const proc = spawnSync(process.execPath, args, {encoding: 'utf8'});
-assert(/Debugger listening on/.test(proc.stderr));
+var procStderr = '';
+var agentStdout = '';
+var needToSpawnAgent = true;
+var needToExit = true;
+
+const procArgs = [`--debug-brk=${common.PORT}`, '-e', '0'];
+const proc = spawn(process.execPath, procArgs);
+proc.stderr.setEncoding('utf8');
+
+const exitAll = common.mustCall((processes) => {
+  processes.forEach((myProcess) => { myProcess.kill(); });
+});
+
+proc.stderr.on('data', (chunk) => {
+  procStderr += chunk;
+  if (/Debugger listening on/.test(procStderr) && needToSpawnAgent) {
+    needToSpawnAgent = false;
+    const agentArgs = ['debug', `localhost:${common.PORT}`];
+    const agent = spawn(process.execPath, agentArgs);
+    agent.stdout.setEncoding('utf8');
+
+    agent.stdout.on('data', (chunk) => {
+      agentStdout += chunk;
+      if (/connecting to .+ ok/.test(agentStdout)) {
+        if (needToExit) {
```

Comment:
Suggestion: fold this into the previous if statement like you do with the outermost if statement.  Saves a level of indent.